src/steps/create-registries/pass-component-name-to-base-component.ts

```diff
@@ -23,12 +23,11 @@ export function passComponentNameToBaseComponent(
 
   const ast = traverse(file, {
     visitClassDeclaration(path) {
-      if (!path.node.superClass) {
-        return false;
-      }
-
-      // @ts-ignore: Assume that types from external packages are correct
-      if (path.node.superClass.name !== baseComponentName) {
+      if (
+        !path.node.superClass ||
+        path.node.superClass.type !== 'Identifier' ||
+        path.node.superClass.name !== baseComponentName
+      ) {
```

Comment:
My impression of `recast` is, its type-check is often accurate and encourages end-developers to practice early exits.